Custom screen size

cole-maclean opened the pull request:
Ok here's my go at providing an optional argument to allow custom screen size scaling requested in #40.

The optional parameter simply abstracts the hard coded screen scaling constant here: https://github.com/deepmind/pysc2/blob/master/pysc2/lib/renderer_human.py#L139 into a user adjustable parameter. I've updated play.py and agent.py to accept --window_scaling as an argument and this flows through to the RenderHuman object and is passed in as a parameter to the _get_max_window_size function to be used in the above line. I don't think there should be any gotchyas, but let me know if you have any questions. I added a note about the argument in the environment.md docs.

Thanks @tewalds and DeepMind for the sweet repo, looking forward to playing with it.

-Cole

cole-maclean commented:
I signed it!

tewalds commented:
When does the existing 0.75 fail? Is that wrong on some resolution, eg two monitors?

cole-maclean commented:
Yeah, it's in reference to issue #40 where they seem to have multiple monitors, but I'm using my single laptop monitor (3840x2160), and the rendered window is about 4x's larger than my display. I'm not sure the root cause of why 0.75 with your screen size calculations don't work everywhere, but likely has to do with the various setups and OSs people use. This argument allows users to pass in a scaling that'll work for them no matter their setup, though it might be better to house this in a config file somewhere so it doesn't need to be passed as an argument every time.

tewalds commented:
Hmm, a bit of research suggests that maybe this is related to hidpi where resolution and window size are defined differently. I don't have a hidpi screen, so could you try out some of the functions in https://www.pygame.org/docs/ref/display.html#pygame.display.Info to see if there's a way to make it calculate the correct size with and outwith hdpi?

cole-maclean commented:
Sure, I'll take a look, would definitely be less clunky if it just directly calcs an appropriate screen size for all setups.

cole-maclean commented:
Ok here we go. I rebased this branch on the current master, so the previous approach was cleaned up. 

I found a solution here: https://stackoverflow.com/questions/27421391/pygame-display-info-giving-wrong-resolution-size. Key was to add these two lines to the renderer_human.py module:

```
import ctypes
ctypes.windll.user32.SetProcessDPIAware()
```
Works like a charm on my system, and way cleaner than my original custom argument solution. Only catch is how to test this on other systems as this is a Windows only fix. I added a check to ensure the above line only fires if the system is Windows based.

Take a look and let me know if there are any other issues.

Thanks,
-Cole

tewalds commented:
Merged internally. It'll be in the next release.
